landlab/components/radiation/radiation.py

```diff
@@ -185,26 +239,51 @@ def __init__(
 
         self._nodal_values = self._grid["node"]
         self._cell_values = self._grid["cell"]
+
         self._slope, self._aspect = grid.calculate_slope_aspect_at_nodes_burrough(
             vals="topographic__elevation"
         )
 
         self._cell_values["Slope"] = self._slope
         self._cell_values["Aspect"] = self._aspect
 
+        # all closed node values will be set to -9999 to be marked
+        # for non consideration later.
+        closed_nodes = self._grid.status_at_node == self._grid.BC_NODE_IS_CLOSED
+        self._nodal_values["topographic__elevation"][closed_nodes] = -9999
```

Comment:
Can do, but for this we need to add a field to the nodal values populated with the node statuses, then map that field to cells using the map_node_to_cell function (because _radf is a cell value and so is elevation). This kind of fix works perfectly fine though so I will be making this change